web_src/js/utils.js

```diff
@@ -133,3 +133,9 @@ export function convertImage(blob, mime) {
     }
   });
 }
+
+const {appSubUrl} = window.config;
+
+export function getCurAbsUrl(relUrl) {
+  return `${window.location.origin}${appSubUrl}${relUrl}`;
+}
```

Comment:
This function does wrong.

* It should behave like a browser:
    * if it gets an absolute URL, just returns it
    * if it gets an relative URL, prepend the origin (but do not touch appSubUrl)

The caller should provide a relative URL to it, the backend should already prepared the AppSubURL for the path.

Reply:
Done.